src/datatypes.ts

```diff
@@ -87,14 +70,41 @@ interface ILogger {
    * @return {ILogLevel} the current logging level
    */
   getLevel(): ILogLevel;
-   /**
+
+  time(label: string): void;
+  timeEnd(label: string): void;
+
+  enabledFor(level: ILogLevel): boolean;
+}
+
+export interface IJSLoggerDefaultsType {
```

Comment:
I think this would be a good oppertunity to make the `LogLevel`'s a publicly exported enum, we can do this in a subsequent refactor if you want to reduce the amount of change in this PR.

Reply:
Yeah, using TypeScript enums would be awesome :)

I'll see if we can utilize them in a way that would still keep them on the `IJSLoggerDefaultsType`, such that the tests pass without any modifications.

Reply:
@jonnyreeves I think that would work. Would you be fine with a string based enum to preserve the human readability when it puts up during debugging? Put follow-up PR might be best, I hope no-one is using the internals of the log level interface.